packages/expo-crypto/android/src/main/java/expo/modules/crypto/CryptoModule.java

```diff
@@ -0,0 +1,56 @@
+package expo.modules.crypto;
+
+import android.content.Context;
+import android.util.Base64;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import expo.core.ExportedModule;
+import expo.core.ModuleRegistry;
+import expo.core.interfaces.ExpoMethod;
+import expo.core.interfaces.ModuleRegistryConsumer;
+import expo.core.Promise;
+
+public class CryptoModule extends ExportedModule implements ModuleRegistryConsumer {
+
+  public CryptoModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+  }
+
+  @Override
+  public String getName() {
+    return "ExpoCrypto";
+  }
+
+  @ExpoMethod
+  public void digestStringAsync(String algorithm, String data, final Map<String, Object> options, final Promise promise) {
+    // hex
+    String encoding = (String)options.get("encoding");
+
+    try { 
+      MessageDigest md = MessageDigest.getInstance(algorithm);
```

Comment:
Move everything except this out of the try block. Put only the code you think could throw an exception in the try block.